test/TestApps/ActivationTestApp/main.cpp

```diff
@@ -0,0 +1,132 @@
+            AsyncStatus const asyncStatus)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                auto channelUri = result.Channel().Uri();
+
+                std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl << std::endl;
+
+                auto channelExpiry = result.Channel().ExpirationTime();
+
+                // Register Push Event for Foreground
+                pushToken = result.Channel().PushReceived([&](const auto&, PushNotificationReceivedEventArgs args)
+                    {
+                        auto payload = args.Payload();
+
+                        // Do stuff to process the raw payload
+                        std::string payloadString(payload.begin(), payload.end());
+
+                        time(&ltime);
+                        ctime_s(buf, sizeof buf, &ltime);
+                        std::cout << "Push notification content received from FOREGROUND: " << buf << payloadString << std::endl << std::endl;
+
+                        args.Handled(true);
+                    });
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                LOG_HR_MSG(result.ExtendedError(), "We hit a critical non-retryable error with channel request!");
+            }
+        });
+}
+
+int main()
+{
+    time(&ltime);
+    ctime_s(buf, sizeof buf, &ltime);
+
+    std::cout << "Project Reunion Push Notification Test App: " << buf << std::endl;
+
+    PushNotificationActivationInfo info(
+        PushNotificationRegistrationKind::PushTrigger | PushNotificationRegistrationKind::ComActivator,
+        winrt::guid("ccd2ae3f-764f-4ae3-be45-9804761b28b2")); // same clsid as app manifest
+
+    auto token = PushNotificationManager::RegisterActivator(info);
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+    if (kind == ExtendedActivationKind::Push)
+    {
+        PushNotificationReceivedEventArgs pushArgs = args.Data().as<PushNotificationReceivedEventArgs>();
+
+        // Call GetDeferral to ensure that code runs in low power
+        auto deferral = pushArgs.GetDeferral();
+
+        auto payload = pushArgs.Payload();
+
+        // Do stuff to process the raw payload
+        std::string payloadString(payload.begin(), payload.end());
+
+        time(&ltime);
+        ctime_s(buf, sizeof buf, &ltime);
+        std::cout << "Push notification content received from BACKGROUND: " << buf << payloadString << std::endl << std::endl;
+
+        // Call Complete on the deferral as good practise: Needed mainly for low power usage
+        deferral.Complete();
+    }
+    else if (kind == ExtendedActivationKind::ToastNotification)
+    {
+        std::cout << "ToastNotification received!" << std::endl;
+    }
+    
+    RequestChannel();
```

Comment:
A few problems here. RequestChannel should be called only for Foreground launches. The code here calls it for background launches which is unadvisable.

Reply:
Also would make the App a little UI friendly. For example, in the case of FG launch, you can say: "Press Enter to exit the App Anytime". For the background case, you can directly exit the process after processing the payload. No need to wait on char.